### DISCUSSION TITLE
"Open containing folder" action in asset search results

### DISCUSSION THREAD

**Discussion by Mosnar** [craftcms/cms#14227] [Ideas] [answered]
It'd be nice if there was a way to right-click an asset in the asset index's search results to go to its containing folder. Currently, there isn't an easy way to figure out which folder an asset is in without selecting the asset, opening the edit screen, and viewing the path in the panel to the right. This would also be inline with functionality we see on desktop in Finder: when searching, results get an option called "Show in enclosing folder" in the search results. Clicking this opens a finder window with the item preselected.

**Comment by brandonkelly** [comment]
> Clicking this opens a finder window with the item preselected.

That could be a little tricky if the asset doesn’t live on the first page. I guess we could try to figure out which page it will show up on, but that could be expensive depending on the size of the folder.

**Reply by Mosnar** [reply]
I thought about that as well and couldn't come up with a great solution. One idea I had was to "pin" the item at the top of the list by sorting by its ID, followed by the default sort params.

Regardless, I don't think it's critical to pre-select the item - I'm personally comfortable considering that out of scope for this feature request.

**Reply by brandonkelly** [reply]
Maybe we just have the search input pre-filled-in to the asset’s filename.

**Reply by Mosnar** [reply]
That's a great idea! Could even use the search syntax `filename::"birthday party.mkv"`

**Comment by brandonkelly** [comment]
Related, we just released 5.0.0-beta.4 with a new “Show in folder” asset action (#14482)

<img src="https://github.com/craftcms/cms/assets/47792/7f279820-006c-40f0-889b-6c02c21aff50" alt="An asset’s action menu, with “Show in folder” highlighted" width="314" height="346">

(Bulk element action will be added shortly…)

**Comment by brandonkelly** [comment, marked as answer]
Alright, the bulk element action is now available in Craft 5.0.0-beta.5 🎉 

It will only show up when searching across subfolders.


https://github.com/craftcms/cms/assets/47792/d9f5b8b7-8525-46c9-8712-e7d809d02969
